src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -41,8 +47,11 @@ export const createSavedQueryService = (
       query,
     };
 
+    let references = [];
```

Comment:
i would extract this into `common/query/saved_query/persistable_state` as it needs to be exposed on the server as well.

Reply:
or common/query/persistableState (where filters utilities already are)

Reply:
and please expose on the server as well.

Reply:
Would it make sense to create a dedicated service with routes on the server? And then the saved query service on the client would just make HTTP calls and this logic would all be done on the server instead? I'm starting to go down this path, let me know if you disagree with this approach.